lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -129,14 +129,42 @@ def generate_refresh_token
     end
 
     def generate_token
+      set_timestamps
+
       generator = Doorkeeper.configuration.access_token_generator.constantize
-      self.token = generator.generate(resource_owner_id: resource_owner_id,
-                                      scopes: scopes, application: application,
-                                      expires_in: expires_in)
+      self.token = generator.generate(
+        resource_owner_id: resource_owner_id,
+        scopes: scopes,
+        application: application,
+        expires_in: expires_in,
+        created_at: created_at
+      )
     rescue NoMethodError
       raise Errors::UnableToGenerateToken, "#{generator} does not respond to `.generate`."
     rescue NameError
       raise Errors::TokenGeneratorNotFound, "#{generator} not found"
     end
+
+    def set_timestamps
+      if created_at.nil?
+        self.created_at = current_time
+
+        if respond_to? :updated_at=
```

Comment:
> The Active Record model does not have an updated_at column but the Mongoid model does.

Do you personally use mongo? I'd say that tokens are never update and should not have this column in any ORM (or have it as an extension); or that indeed we do need this data, in which case it should be incorporated into doorkeeper ("main").